### PR TITLE
New VPN provider support, bug fixes

### PR DESCRIPTION
**Pull request by lianke123321**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -18,10 +18,14 @@
...
                       "dnspython >= 1.11.0",
-                      "requests >= 2.5.1",
+                      "requests >= 2.9.1",
                       "trparse >= 0.2.1",
                       "pycurl >= 7.19.5",
-                      "urllib3 == 1.9.1"],
+                      "urllib3 == 1.9.1",
+                      "dnspython == 1.12.0",
+                      "M2Crypto == 0.21.1",
+                      "BeautifulSoup == 3.2.1",
```
Comment (rpanah) [inline setup.py]: What is this? Is it used by the main Centinel package/experiments?

File: setup.py [hunk 2]
```diff
@@ -18,10 +18,14 @@
...
     install_requires=["argparse >= 1.2.1",
                       "dnspython >= 1.11.0",
-                      "requests >= 2.5.1",
+                      "requests >= 2.9.1",
                       "trparse >= 0.2.1",
                       "pycurl >= 7.19.5",
-                      "urllib3 == 1.9.1"],
+                      "urllib3 == 1.9.1",
+                      "dnspython == 1.12.0",
+                      "M2Crypto == 0.21.1",
```
Comment (rpanah) [inline setup.py]: Remove this. This one is optional (TLS will be disabled if this isn't installed).

File: setup.py (hunk 1 above)
Comment (lianke123321) [inline setup.py]: These are used by scripts for downloading ipvanish and purevpn configs.

Comment (rpanah) [inline setup.py]: Then it doesn't belong here.

File: setup.py (hunk 2 above)
Comment (lianke123321) [inline setup.py]: Okay

File: setup.py (hunk 1 above)
Comment (lianke123321) [inline setup.py]: So where does it belong?

Comment (rpanah) [inline setup.py]: In the documentation of VPN.

**Pull request merged**